File: Leetcode-Problems/Minimum Distance Between Three Equal Elements I.cpp
class Solution
{
public:
    int minimumDistance(vector<int> &nums)
    {
        int ans = INT_MAX;
        bool found = false;
        for (int i = 0; i < nums.size(); i++)
        {
            for (int j = i + 1; j < nums.size(); j++)
            {
                for (int k = j + 1; k < nums.size(); k++)
                {
                    if (nums[i] == nums[j] && nums[j] == nums[k])
                    {
                        found = true;
                        ans = min(ans, abs(i - j) + abs(j - k) + abs(k - i));
                    }
                }
            }
        }
        if (found)
            return ans;
        return -1;
    }
};
©leetcode

File: Leetcode-Problems/Minimum Distance Between Three Equal Elements I.java
class Solution {
    public int minimumDistance(int[] nums) {
        HashMap<Integer, List<Integer>> mp = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            mp.computeIfAbsent(nums[i], t -> new ArrayList<>()).add(i);
        }
        int ans = -1;
        for (Map.Entry<Integer, List<Integer>> entry : mp.entrySet()) {
            List<Integer> temp = entry.getValue();
            if (temp.size() < 3)
                continue;
            for (int i = 0; i < temp.size() - 2; i++) {
                int dist = Math.abs(temp.get(i + 2) - temp.get(i + 1)) + Math.abs(temp.get(i + 1) - temp.get(i))
                        + Math.abs(temp.get(i) - temp.get(i + 2));
                if (ans == -1)
                    ans = dist;
                ans = Math.min(ans, dist);
            }
        }
        return ans;
    }
}